src/api/controllers/TeamController.js

```diff
@@ -120,6 +121,78 @@ module.exports = {
       });
   },
 
+  getAll: (_, res) => {
+    var response = [];
+
+    Team.find()
+      .sort({ dateCreated: -1 })
+      .populate("categories")
+      .then((teams) => {
+        teams.forEach((team) => {
+          toSanitize = ["__v", "followers", "dateCreated", "owner"];
+          team = sanitize(team, toSanitize);
+          team.categories = team.categories.map((c) => c.name);
+
+          response.push(team);
+        });
+      })
+      .then(() => {
+        return res.send(response);
+      });
+  },
+
+  getTeam: (req, res) => {
+    Team.findOne({ _id: req.params.team_id })
+      .populate("categories")
+      .then((team) => {
+        if (team) {
+          toSanitize = ["__v", "owner", "dateCreated"];
+          team = sanitize(team, toSanitize);
+          team.categories = team.categories.map((c) => c.name);
+          team.followers = team.followers.length;
+
+          Action.find({ organizer: team._id })
+            .sort({ date: -1 })
+            .populate("categories")
+            .then((actions) => {
+              team["upcoming"] = [];
+              team["past"] = [];
+
+              actions.forEach((action) => {
+                satinizeAction = [
+                  "__v",
+                  "attendees",
+                  "saves",
+                  "dateCreated",
+                  "organizer",
+                ];
+
+                action = sanitize(action, satinizeAction);
+                action.categories = action.categories.map((c) => c.name);
+
+                if (action.date > new Date()) {
```

Comment:
Extract this Date object outside of the forEach loop, since we don't need to create a new one in every iteration.